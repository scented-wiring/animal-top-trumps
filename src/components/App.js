import "../styles/App.css";
import Home from "../components/Home";
import CreateCards from "../components/CreateCards";
import { Link, Switch, Route } from "react-router-dom";
import logo from "../dog.png";

const App = () => {
  return (
    <div className="App">
      <div className="title">
        <Link to="/">
          <img id="logo" src={logo} alt="Logo" width="70px" height="70px" />{" "}
        </Link>
        <Link to="/">
          <h1>Animal Top Trumps</h1>
        </Link>
      </div>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/createcards" component={CreateCards} />
      </Switch>
    </div>
  );
};

export default App;
