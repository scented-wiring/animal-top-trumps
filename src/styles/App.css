.App {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  background-color: #00d7b0;
}

.title {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #73ed8f;
  border-bottom: 3px black solid;
  letter-spacing: 3px;
  transition: all 0.4s ease 0s;
  padding: 15px 0 15px 0;
}

.title:hover {
  color: white;
  transition: all 0.4s ease 0s;
}

#logo {
  padding: 10px;
  margin-right: 15px;
  border: 3px solid black;
  border-radius: 50%;
  background-color: white;
}

button {
  align-self: center;
  height: 35px;
  width: 100px;
  background-color: #73ed8f;
  margin-top: 10px;
  font-family: "Staatliches", sans-serif;
  font-size: 20px;
  border: 3px black solid;
  transition: all 0.4s ease 0s;
}

button:hover {
  background: black;
  color: white;
  transition: all 0.4s ease 0s;
}

a {
  color: inherit;
  text-decoration: none;
}

h1 {
  font-size: 60px;
  font-family: "Staatliches", sans-serif;
  margin: 0;
}

h2 {
  font-size: 35px;
  font-family: "Staatliches", sans-serif;
  transition: all 0.4s ease 0s;
  margin: 20px 0 20px 0;
}

h3 {
  margin: 0;
  font-size: 30px;
  font-family: "Staatliches", sans-serif;
  transition: all 0.4s ease 0s;
  margin-top: 20px;
}

h3:hover {
  transition: all 0.4s ease 0s;
  color: white;
}
