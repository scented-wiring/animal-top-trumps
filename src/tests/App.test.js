import { render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";

describe("App component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });

  test("renders the page title", () => {
    const { getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(getByText("Animal Top Trumps")).toBeInTheDocument();
  });

  test("renders two home links", () => {
    const { getAllByRole } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(getAllByRole("link")).toBeTruthy();
    expect(getAllByRole("link")[0]).toHaveAttribute("href", "/");
    expect(getAllByRole("link")[1]).toHaveAttribute("href", "/");
  });

  test("switches components", () => {
    const { getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    fireEvent.click(getByText("Browse Cards"));
    expect(getByText("Loading...")).toBeTruthy();
  });
});
