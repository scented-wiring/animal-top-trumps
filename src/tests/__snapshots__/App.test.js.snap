// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `[Function]`;

exports[`renders correctly 1`] = `[Function]`;
